src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1108,13 +1111,54 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		if (!this._accessibilityBuffer) {
 			return;
 		}
-		const accessibleBufferContentEditable = this._configurationService.getValue(TerminalSettingId.AccessibleBufferContentEditable);
+		// see https://github.com/microsoft/vscode/issues/173532
+		const accessibleBufferContentEditable = isLinux ? 'on' : this._configurationService.getValue(TerminalSettingId.AccessibleBufferContentEditable);
 		this._accessibilityBuffer.contentEditable = accessibleBufferContentEditable === 'on' || (accessibleBufferContentEditable === 'auto' && !this._accessibilityService.isScreenReaderOptimized()) ? 'true' : 'false';
 		// The viewport is undefined when this is focused, so we cannot get the cell height from that. Instead, estimate using the font.
 		const font = this.xterm.getFont();
 		const lineHeight = font?.charHeight ? font.charHeight * font.lineHeight + 'px' : '';
 		this._accessibilityBuffer.style.lineHeight = lineHeight;
+		const commands = this.capabilities.get(TerminalCapability.CommandDetection)?.commands;
+		if (!commands?.length) {
+			const noContent = document.createElement('div');
+			const noContentLabel = nls.localize('terminal.integrated.noContent', "No terminal content available for this session.");
+			noContent.textContent = noContentLabel;
+			this._bufferElementFragment.replaceChildren(noContent);
+			this._accessibilityBuffer.focus();
+			return;
+		}
+		let header;
+		let replaceChildren = true;
+		for (const command of commands) {
+			header = document.createElement('h2');
+			// without this, the text area gets focused when keyboard shortcuts are used
+			header.tabIndex = -1;
+			header.textContent = command.command.replace(new RegExp(' ', 'g'), '\xA0');
+			const output = document.createElement('div');
+			// without this, the text area gets focused when keyboard shortcuts are used
+			output.tabIndex = -1;
+			output.textContent = command.getOutput()?.replace(new RegExp(' ', 'g'), '\xA0') || '';
+			if (replaceChildren) {
+				this._bufferElementFragment.replaceChildren(header, output);
+				replaceChildren = false;
+			} else {
+				this._bufferElementFragment.appendChild(header);
+				this._bufferElementFragment.appendChild(output);
+			}
+		}
+		if (header) {
+			// focus the cursor line's header
+			header.tabIndex = 0;
+		}
```

Comment:
Can we move this stuff into `_provideBufferElements`'s implementation? Consider the user manually focusing the a11y buffer somehow, we should give xterm.js an up to date view when it requests it.

Reply:
Seems like quite an edge case that can be deferred

Reply:
https://github.com/microsoft/vscode/issues/173734

Reply:
Thanks